### ISSUE TITLE
Enable multi-line support for message replies.

### ISSUE THREAD

**Issue by InfamousVague** [Satellite-im/WarpGUI#40]
This will probably just involve copying the Javascript inject from the compose component and copying the behavior over on the message replies. We should probably also set a max line count of 4-5 to prevent the UI looking horrible. 

<img width="863" alt="image" src="https://user-images.githubusercontent.com/1770198/192868900-0eb805c6-210a-4e96-9439-6b1db1b85e5f.png">

vs.

<img width="599" alt="image" src="https://user-images.githubusercontent.com/1770198/192868966-f481ab77-386b-4485-bc1f-354580b37f7d.png">

**Comment by sdwoodbury** [comment]
per Matt's request I tried  to implement the resizing without JavaScript, but was unable to. I tried the following

#### Use `rows` to resize a `textarea`
This didn't work because in Dioxus i didn't have a way to get the width of the `textarea`. If I had the width I could calculate the necessary number of rows by dividing the text length by the width of the `textarea`. 

#### overlay a `div` with a `textarea`
If I overlay a `div` with a `textarea` and set `display: hidden` and then duplicate the `textarea` input in the `div`, then theory the `div` would grow as needed and force the `textarea` to do the same. 

#### Use a `div` with `contenteditable`
I may have found a bug in Dioxus here. see the comments in the minimal reproducible example
```
use dioxus::prelude::*;
use dioxus_html::KeyCode;

fn main() {
    dioxus::desktop::launch(app);
}

fn app(cx: Scope) -> Element {
    let text = use_state(&cx, || String::from(""));
    cx.render(rsx! (
       main {
        p {
            "some text"
        }
        div {
            contenteditable: "true",
            "text_align": "left",
            onkeypress: move |evt| {
                if evt.key_code == KeyCode::Enter  {
                    println!("text submitted: {}", &text);
                    // fails to update the UI. enter "abcd" and it outputs "abcdabcaba"
                    text.set(String::from(""));
                }
            },
            // maybe oninput is only supposed to work on input and textarea elements.
            oninput: move |evt| {
                // doesn't quite work until the content is deleted
                text.set(evt.value.clone());
            },
            // if this next line is commented out, the input works. but there's still not a way to clear the text
            "{text}"
        }
        p {
            "more text"
        }
       }
    ))
}
```

**Comment by sdwoodbury** [comment]
I should be able to use a `textarea` and resize it using this technique: https://stackoverflow.com/questions/17731083/how-to-autogrow-a-textarea-with-css

but I can't clear the textarea. I filed a bug with Dioxus: https://github.com/DioxusLabs/dioxus/issues/597 
I don't know where to start to fix it - the implementation of `value` looks the same for `input` and `textarea`, at least within `html/src/element.rs`

until this gets fixed, I think the JavaScript has to stay.

**Comment by sdwoodbury** [comment]
I got some help from DioxusLabs and was able to clear the TextArea without JavaScript. I was not able to figure out how to resize the TextArea without JavaScript (i could resize it horizontally without JS but we need vertical resizing). 

However, I'm hoping that not using JS to clear the content of the TextArea helps with the double-send bug.

**Comment by dariusc93** [comment]
Resolved in #126
